static/js/services/message-contacts.js

```diff
@@ -12,7 +12,7 @@ var async = require('async'),
     function(HttpWrapper, BaseUrlService) {
       return function(params, callback) {
         var url = BaseUrlService() + '/message_contacts';
-        HttpWrapper.get(url, { params: params })
+        HttpWrapper.get(url, { params: params, timeout: false })
```

Comment:
Unfortunately this introduces the exact issue that the timeout was added to solve. MessageContactsRaw is used indirectly to populate the LHS and RHS of the messages tab. This request must not be allowed to complete after a navigation change.